### PR TITLE
Double-Trouble

### PR DESCRIPTION
**Pull request by Playyne**
Adds a fire-mode that fires both barrels "at once" to the double-barreled shotgun, based on the automatics burst-fire.

:cl:
 * rscadd: The Bartender's shotgun now has a new fire-mode that lets it fire both barrels at once.

### CODE REVIEW

**Comment by Reviire** [comment]
120-180 damage weapon
nice

**Comment by Irradiation** [comment]
Might want to like, put in a drawback to doing this, because unlike both CM servers, we don't have recoil or accuracy variables.

**Comment by Playyne** [comment]
@Irradiation 
That 5% chance is to blow it up when doing this with a faulty weapon.
I thought that the hassle of reloading after every shot and being open to attack while fumbling shells would be some kind of balance factor, but I guess not.
Should it damage your arm because of the recoil?
Should it give a cooldown on trying to open the gun?

**Comment by Irradiation** [comment]
You know what the Obrez has to "discourage" people from using it like a better version of a mosit nugget? It has a 15% to outright drop the gun on the floor. For a gun that may very likely instantly crit somebody, I suggest skyrocketing that chance to _at least_ 50%, while keeping the chance to misfire. If you're feeling like an extra madman, make it **break the arm** that'S firing the gun.

**Comment by Playyne** [comment]
@Irradiation 
I like your style.
I personally thought about adding brute damage to the hand the shotgun is in to the "drops shit due to pain" threshold. That would make the drop chance 100%, after a short delay.

Breaking the arm would discourage people from using the fire-mode outright because it's a pain in the ass to fix.
Why would I add something if I made it so it'll never be used?

**Comment by Irradiation** [comment]
Considering you're basically giving the bartender shotgun the ability to crit somebody in effectively 1 shot, you damn better put a solid drawback to it. As for broken bones pain, eat some Oxycodone and tell me if you can still feel those 8 broken bones you have.

**Comment by Playyne** [comment]
I'll go work on some !FUN! drawbacks for this.

**Comment by Reviire** [comment]
I don't think there's any suitable drawbacks, bar high inaccuracy. Fine for buckshot (which wouldn't be THAT bad), but will do some pretty serious cucking for slugs (Which will not only oneshot crit, but oneshot kill because of head gibbing).

A handheld weapon that can kill in a single shot, is still a handheld weapon that can kill in a single shot. Even if it blows off your hand. At least tasers let you scream first.

**Comment by Irradiation** [comment]
For reference, another gun that can one-shot people is the Hecate sniper, which is only obtained through bussing, vaults, Deathsquad prep room or a wizard summoning guns. You also only get 1 round of it, it knocks you on your ass if you aren't buckled and temporary renders you deaf after shooting. It also _only_ deals 120~ brute (after penetrating walls).

**Comment by AutismSpider** [comment]
Should Break the hand if you dont have the strong/buff mut

**Comment by Playyne** [comment]
The main issue with this PR is that shotguns and their ammo are balanced around slow firing, which this PR breaks completely, the only way to remotely make this balanced would be to lower the slug damage to 40, which would be a huge nerf to an already underused class of weapons.
The AR and the Syndicate C20-R, from which I based this PR, both fire 20 damage bullets with a 4 round burst, which will certainly fuck you up good but it's no instacrit like this.
I was currently working on adding several randomly picked drawbacks including stuns that throw you on your ass, breaking your arm, making the gun fly out of your hand like you're a college girl first-time shooter, a jamming mechanic and so on.
If someone has any better balance suggestions, please post them, if I don't get anything soon I'll just axe this meme of a PR and work on something slightly less retarded.

**Comment by AutismSpider** [comment]
@Playyne the barman is lacking antag items COULD turn this into shotgun he can pay TC for?

**Comment by Playyne** [comment]
@AutismSpider Really good point, I could rebuild this into a modification kit that turns the shotgun into one that lets you do this, double-barrel shotguns usually don't let you fire both barrels at once unless it's the kind with two triggers that can be pulled simultaneously.
This would still put you in instacrit though without allowing any screams for help, if this gets merged it would certainly enter the ranks of sleepy-pen-C4 shittery.

**Comment by Playyne** [comment]
@AutismSpider I'll keep the idea in mind but I'll wait for some more feedback from other players.

**Comment by SonixApache** [comment]
Just make the shells have a 60% spread or something, so the chance of two slugs hitting the same target (except at point blank) is really low.

That or we could just change the double-barrel into a 20 gauge or something

**Comment by Playyne** [comment]
@SonixApache 
I don't see an easy way in adding spread to projectiles, guns like the gatling work around that, will need to look into it.
Though adding a completely new ammo type for one (1) feature in one (1) gun would be excessively bloaty.

**Comment by Playyne** [comment]
Added 4 randomly selected drawbacks:
* The shells get stuck in the barrels, needing to be manually pushed out by a rod
* The gun flies out of your grip and launches itself 2 tiles behind you
* The gun throws you on your ass for 3 seconds
* The gun deals random (15-40) brute damage to your arm, which may or may not break it
The odds are subject to change by feedback.
The gun is still 100% accurate though.

**Comment by Blithering** [comment]
Maybe it would be a good idea to have a check to see if the shotgun is sawn off and increase the severity of the backfire if so

**Comment by Playyne** [comment]
@Blithering
The sawn-off is pretty half-assedly shortened, it just has the shortened barrel, the stock is still fully intact.
So it doesn't make sense realistically to make its drawbacks much more severe if you can still properly brace against it, and sawn-offs do not have higher recoil by having a shorter barrel.
Gameplay-wise the sawn-off can be hidden in belt loops and backpacks, so with this new firemode it will be easier to pull it out from somewhere unseen and surprise someone, so balancing could be appropriate.

Honestly, I have implemented all I really wanted to implement with this PR, I'm not a fan of random weapon spread. If a maintainer posts a "do-or-die" issue I have to fix, I will, but until then I don't really want to add anything to this.

File: code/modules/projectiles/guns/projectile/shotgun.dm
Comment (ShiftyRail) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: You could probably tell what those numbers actually do or use defines.
```dm
#define JAMMED 1
#define RECOIL 2
#define KNOCKDOWN 3
#define BREAK_ARM 4
var/severity = pick(JAMMED, ...)
switch(severity)
         if (JAMMED)
               // ...
```
But that's not a major concern, just a little bit of readability

Comment (Playyne) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: Well, the recoil effect selection was concentrated in only a dozen lines so an attentive reader could've figured it out easily, especially when reading the comments I added to the code.
More readability can't hurt though, so I did it anyway.

File: code/modules/projectiles/guns/projectile/shotgun.dm
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: Ew. Use spawn(1 SECONDS) instead.

File: code/modules/projectiles/guns/projectile/shotgun.dm
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: No need for this.

File: code/modules/projectiles/guns/projectile/shotgun.dm
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: If this is only needed for one thing that's behind a prob check, why not have it defined down there?

File: code/modules/projectiles/guns/projectile/shotgun.dm
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: No need for the null, null, null chain. You can underload function arguments in byond.

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

**Comment by Playyne** [comment]
Jesus Christ what the fuck.

**Comment by SonixApache** [comment]
ya blew it

never rebase off bleeding-edge

**Pull request closed without merging**
